test/unit/conftest.py

```diff
@@ -196,6 +190,17 @@ def rpcq_server() -> rpcq.Server:
     return rpcq.Server()
 
 
+@pytest.fixture(scope="function")
```

Comment:
New fixture, now that I know that tests might run in parallel. It's not 100% bullet-proof, but it has mitigated the flakes drastically. (A flake used to occur on every test run -- now, I haven't seen a flake yet after a handful of runs in a row).